### PR TITLE
Bump recommended ESP32 IDF to 4.4.6

### PR DESCRIPTION
**Pull request by bdraco**
# What does this implement/fix?

ESP-IDF https://github.com/espressif/esp-idf/releases/tag/v4.4.6 

Lots of BLE disconnect fixes

related issue https://github.com/esphome/issues/issues/5119

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [x] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -104,7 +104,7 @@ extra_scripts = post:esphome/components/esp8266/post_build.py.script
 ; This are common settings for the ESP32 (all variants) using Arduino.
 [common:esp32-arduino]
 extends = common:arduino
-platform = platformio/espressif32@5.4.0
+platform = platformio/espressif32@6.5.0
```
Comment (bdraco) [inline platformio.ini]:
```suggestion
platform = platformio/espressif32@5.4.0
```

File: platformio.ini [hunk 2]
```diff
@@ -134,9 +134,9 @@ extra_scripts = post:esphome/components/esp32/post_build.py.script
 ; This are common settings for the ESP32 (all variants) using IDF.
 [common:esp32-idf]
 extends = common:idf
-platform = platformio/espressif32@5.4.0
+platform = platformio/espressif32@6.5.0
```
Comment (bdraco) [inline platformio.ini]:
```suggestion
platform = platformio/espressif32@5.4.0
```

File: esphome/components/esp32/__init__.py [hunk 3]
```diff
@@ -237,7 +237,7 @@ def _arduino_check_versions(value):
     value = value.copy()
     lookups = {
         "dev": (cv.Version(2, 1, 0), "https://github.com/espressif/arduino-esp32.git"),
-        "latest": (cv.Version(2, 0, 9), None),
+        "latest": (cv.Version(2, 0, 14), None),
```
Comment (bdraco) [inline esphome/components/esp32/__init__.py]:
```suggestion
        "latest": (cv.Version(2, 0, 9), None),
```

**Comment by bdraco** [comment]
Everything running smoothly after a few hours. I did all my esp32 devices

**Review by jesserockz — APPROVED** [review]

**Comment by bdraco** [comment]
thanks

**Pull request merged**